editor/src/components/editor/store/dispatch.tsx

```diff
@@ -761,14 +782,22 @@ function removeNonExistingViewReferencesFromState(editorState: EditorState): Edi
   }
 }
 
-function filterNonExistingViews(
-  allPaths: Array<TemplatePath>,
-  views: Array<TemplatePath>,
-): Array<TemplatePath> {
-  const filtered = views.filter((path) => TP.containsPath(path, allPaths))
-  if (filtered.length !== views.length) {
-    return filtered
-  } else {
-    return views
+function findMatchingTemplatePath(
+  oldAllPaths: Array<TemplatePath>,
+  newAllPaths: Array<TemplatePath>,
+  pathToUpdate: TemplatePath,
+): [TemplatePath] | [] {
+  const pathStillExists = newAllPaths.findIndex((p) => TP.pathsEqual(p, pathToUpdate)) > -1
+  if (pathStillExists) {
+    return [pathToUpdate]
   }
+  const parentPath = TP.parentPath(pathToUpdate)
+  const parentPathStillExists = newAllPaths.findIndex((p) => TP.pathsEqual(p, parentPath)) > -1
+  if (parentPathStillExists) {
+    const oldChildren = oldAllPaths.filter((p) => TP.isChildOf(p, parentPath))
+    const oldChildIndex = oldChildren.findIndex((p) => TP.pathsEqual(p, pathToUpdate))
+    const newChildren = newAllPaths.filter((p) => TP.isChildOf(p, parentPath))
+    const potentialNewPath = newChildren[oldChildIndex]
+    return potentialNewPath == null ? [] : [potentialNewPath]
+  } else return []
```

Comment:
Can you wrap this all in an `else` block please

Reply:
whoa who wrote this code